src/test/java/org/agrona/agent/BufferAlignmentAgentTest.java

```diff
@@ -0,0 +1,320 @@
+package org.agrona.agent;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.function.IntConsumer;
+
+import org.agrona.BitUtil;
+import org.agrona.DirectBuffer;
+import org.agrona.ExpandableArrayBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.AtomicBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import net.bytebuddy.agent.ByteBuddyAgent;
+
+public class BufferAlignmentAgentTest
+{
+
+    private static final String TEST_STRING = "BufferAlignmentTest";
+
+    @BeforeClass
+    public static void installAgent()
+    {
+        BufferAlignmentAgent.agentmain("", ByteBuddyAgent.install());
+    }
+
+    @AfterClass
+    public static void removeAgent()
+    {
+        BufferAlignmentAgent.removeTransformer();
+    }
+
+    @Test
+    public void testUnsafeBufferFromByteArray()
+    {
+        testUnsafeBuffer(new UnsafeBuffer(new byte[256]), 0);
+    }
+
+    @Test
+    public void testUnsafeBufferFromByteArrayWithOffset()
+    {
+        final UnsafeBuffer buffer = new UnsafeBuffer(new byte[256], 1, 128);

```

Comment:
Good point, I totally missed than Unsafe.ARRAY_BYTE_BASE_OFFSET is not necessary a multiple of 8. I definitely need to update the heap-based tests to handle this.

However, do you think that all this alignment stuff deserve a better documentation ? Unaligned accesses can be really nasty on some platforms, and such subtle things can easily be missed when trying get correct alignment

Another related concern is the behavior of ExpandableArrayBuffer when Unsafe.ARRAY_BYTE_BASE_OFFSET is not a multiple of 8. Does it make sense to add another offset to make sure that the address of index 0 in the buffer is a multiple of 8 ?

Let me know if you want to create side issues for these concerns.